graphs/gen/duplication.go

```diff
@@ -0,0 +1,121 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gen
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/simple"
+)
+
+// UndirectedMutator is an undirected graph builder that can remove edges.
+type UndirectedMutator interface {
+	graph.UndirectedBuilder
+	graph.EdgeRemover
+}
+
+// Duplication constructs a graph in the destination, dst, of order n. New nodes
```

Comment:
I am not sure that this description is correct/accurate. As I understand it, it suggests that dst will have n nodes after return but in fact it will have |V|+n. Also, it does not construct a graph, it expands an existing graph.